crates/services/consensus_module/poa/src/service.rs

```diff
@@ -567,29 +577,27 @@ where
                             let _ = response.send(result);
                         }
                     }
-                    should_continue = true;
+                    TaskNextAction::Continue
                 } else {
                     tracing::error!("The PoA task should be the holder of the `Sender`");
-                    should_continue = false;
+                    TaskNextAction::Stop
                 }
             }
             _ = next_block_production => {
                 match self.on_timer().await.context("While processing timer event") {
-                    Ok(()) => should_continue = true,
+                    Ok(()) => TaskNextAction::Continue,
                     Err(err) => {
                         // Wait some time in case of error to avoid spamming retry block production
                         tokio::time::sleep(Duration::from_secs(1)).await;
-                        return Err(err);
+                        err.into()
```

Comment:
Another nit for clarity - maybe we shouldn't hide what happens here in the `From` implementation? Maybe it'll be good to explicitly see here that we'll continue running, despite the error (`TaskNextAction::ErrorContinue(err)`).